// @profabric/angular-components/esm2020/lib/web-components-angular.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { DIRECTIVES } from '../generated/directives';
import * as i0 from "@angular/core";
import * as i1 from "../generated/directives/proxies";
export class ProfabricComponentsModule {
}
ProfabricComponentsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.1.3", ngImport: i0, type: ProfabricComponentsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ProfabricComponentsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.1.3", ngImport: i0, type: ProfabricComponentsModule, declarations: [i1.PfButton, i1.PfCheckbox, i1.PfDropdown, i1.PfImage, i1.PfPhoneInput, i1.PfSelect], imports: [CommonModule], exports: [i1.PfButton, i1.PfCheckbox, i1.PfDropdown, i1.PfImage, i1.PfPhoneInput, i1.PfSelect] });
ProfabricComponentsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.1.3", ngImport: i0, type: ProfabricComponentsModule, imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.1.3", ngImport: i0, type: ProfabricComponentsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [...DIRECTIVES],
                    exports: [...DIRECTIVES],
                }]
        }] });
