// @profabric/web-components/dist/components/index.js
import { setMode } from '@stencil/core/internal/client';
export { setAssetPath, setPlatformOptions } from '@stencil/core/internal/client';
export { PfButton, defineCustomElement as defineCustomElementPfButton } from './pf-button.js';
export { PfCheckbox, defineCustomElement as defineCustomElementPfCheckbox } from './pf-checkbox.js';
export { PfDropdown, defineCustomElement as defineCustomElementPfDropdown } from './pf-dropdown.js';
export { PfImage, defineCustomElement as defineCustomElementPfImage } from './pf-image.js';
export { PfPhoneInput, defineCustomElement as defineCustomElementPfPhoneInput } from './pf-phone-input.js';
export { PfSelect, defineCustomElement as defineCustomElementPfSelect } from './pf-select.js';

const Modes = ['bs4', 'bs5'];

setMode((elm) => {
  let mode;
  if (window.PF &&
    window.PF &&
    window.PF.config.mode) {
    mode = window.PF.config.mode;
  }
  if (elm.getAttribute('mode')) {
    mode = elm.getAttribute('mode');
  }
  if (!mode) {
    mode = 'bs5';
  }
  if (!Modes.includes(mode)) {
    throw new Error('Not supported mode');
  }
  elm.setAttribute('mode', mode);
  return mode;
});
