// ng-validate-equal/esm2015/lib/ng-validate-equal.module.js
import { NgModule } from '@angular/core';
import { ValidateEqualDirective } from './ng-validate-equal.directive';
import * as i0 from "@angular/core";
export class ValidateEqualModule {
}
ValidateEqualModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.0", ngImport: i0, type: ValidateEqualModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ValidateEqualModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.0", ngImport: i0, type: ValidateEqualModule, declarations: [ValidateEqualDirective], exports: [ValidateEqualDirective] });
ValidateEqualModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.0", ngImport: i0, type: ValidateEqualModule, imports: [[]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.0", ngImport: i0, type: ValidateEqualModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [ValidateEqualDirective],
                    imports: [],
                    exports: [ValidateEqualDirective]
                }]
        }] });
